third_party/move/move-vm/types/src/utils/union_find.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(Debug)]
+pub struct UnionFind {
```

Comment:
Please add documentation. It is a data structure everyone knows, but it would be great to have a docstring which specifies complexities, use cases, etc.